src/main/java/teammates/common/datatransfer/StudentProfileAttributes.java

```diff
@@ -65,90 +66,97 @@ public StudentProfileAttributes() {
             }
+            return "";
         }
-        
-        return reminder;
     }
-    
+
     private boolean isMultipleFieldsEmpty() {
-        int numEmptyFields = 0;
-        if (this.shortName.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.email.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.nationality.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.moreInfo.isEmpty()) {
-            numEmptyFields ++;
-        }
-        if (this.pictureKey.isEmpty()) {
-            numEmptyFields ++;
-        }
-        
+        int numEmptyFields = returnOneIfEmpty(shortName)
+                + returnOneIfEmpty(email)
+                + returnOneIfEmpty(nationality)
+                + returnOneIfEmpty(moreInfo)
+                + returnOneIfEmpty(pictureKey);
         return numEmptyFields > 1;
     }
 
+    private int returnOneIfEmpty(String str) {
+        return str.isEmpty() ? 1 : 0;
+    }
+
     @Override
     public List<String> getInvalidityInfo() {
         FieldValidator validator = new FieldValidator();
         List<String> errors = new ArrayList<String>();
         String error;
-        
-        error= validator.getInvalidityInfo(FieldValidator.FieldType.GOOGLE_ID, googleId);
-        if(!error.isEmpty()) { errors.add(error); }
-        
+
+        error = validator.getInvalidityInfo(FieldValidator.FieldType.GOOGLE_ID, googleId);
+        if (!error.isEmpty()) {
+            errors.add(error);
+        }
+
         // accept empty string values as it means the user has not specified anything yet.
-        error = shortName.isEmpty() ? "" : validator.getInvalidityInfo(FieldValidator.FieldType.PERSON_NAME, shortName);
-        if(!error.isEmpty()) { errors.add(error); }
-        
-        error = email.isEmpty() ? "" : validator.getInvalidityInfo(FieldValidator.FieldType.EMAIL, email);
-        if(!error.isEmpty()) { errors.add(error); }
-        
-        error = institute.isEmpty() ? "" : validator.getInvalidityInfo(FieldValidator.FieldType.INSTITUTE_NAME, institute);
-        if(!error.isEmpty()) { errors.add(error); }
-        
-        error = nationality.isEmpty() ? "" : validator.getInvalidityInfo(FieldValidator.FieldType.NATIONALITY, nationality);
-        if(!error.isEmpty()) { errors.add(error); }
-        
-        error = validator.getInvalidityInfo(FieldValidator.FieldType.GENDER, gender);
-        if(!error.isEmpty()) { errors.add(error); }
-        
+        error = shortName.isEmpty() ? "" : validator.getInvalidityInfo(
+                FieldValidator.FieldType.PERSON_NAME, shortName);
+        if (!error.isEmpty()) {
+            errors.add(error);
+        }
+
+        error = email.isEmpty() ? "" : validator.getInvalidityInfo(
+                FieldValidator.FieldType.EMAIL, email);
+        if (!error.isEmpty()) {
+            errors.add(error);
+        }
```

Comment:
These code blocks do not match with the intent. The intent is 'if the value is not an empty string and if the value is invalid, record it as an error'. However, the code reads 'if empty string, error is an empty string, ...' and so on. Try to refactor the code to match the intent.